interface/src/pages/transaction/details.tsx

```diff
@@ -97,6 +97,7 @@ export default function TransactionDetails(): JSX.Element {
   return (
     <MForm
       initialValues={ initialValues }
+      enableReinitialize={ true }
```

Comment:
It is a Formik [prop](https://formik.org/docs/api/formik#enablereinitialize-boolean). Now, since the `initialValues` object changes when a Similar Transaction is clicked, we need to let Formik know about this change. Otherwise, the inputs will keep displaying the values from the previous transaction.